subsys/random/Kconfig

```diff
@@ -52,4 +52,14 @@ config XOROSHIRO_RANDOM_GENERATOR
 
 	  It is so named because it uses 128 bits of state.
 
+config XOROSHIRO_DEV_ID_RANDOM_GENERATOR
+	bool "Use Xoroshiro128+ without entropy as PRNG"
+	depends on (DEVICE_ID_HAS_DRIVER && TEST_RANDOM_GENERATOR)
+	help
+	  Enables the Xoroshiro128+ pseudo-random number generator, that
+	  uses the hwinfo device id driver as a seed source. This is not a
+	  cryptographically secure random number generator and it produces a
+	  predictable series of "random" numbers that is the same after every
```

Comment:
I was going to say that I was against this patch until I saw that `CONFIG_TEST_RANDOM_GENERATOR` is required for this to work.  I'm fine with it this way.

Maybe just clarify that two different devices will generate distinct numbers?

Reply:
We can't guarantee that two different devices ALWAYS have different IDs. That's why we renamed it from get unique id to get id. But the probability is very high that they differ from device to device.